zcncore/execute_transactions.go

```diff
@@ -86,14 +116,34 @@ func MinerSCKill(providerId string, providerType Provider, client ...string) (ha
 
 }
 
-func StorageSCCollectReward(providerId string, providerType Provider, client ...string) (hash, out string, nonce int64, txn *transaction.Transaction, err error) {
-	return transaction.SmartContractTxn(StorageSmartContractAddress, transaction.SmartContractTxnData{
+func StorageSCStakePoolLock(providerId string, providerType Provider, lock uint64, fee uint64, client ...string) (hash, out string, nonce int64, txn *transaction.Transaction, err error) {
+	return transaction.SmartContractTxnValueFee(StorageSmartContractAddress, transaction.SmartContractTxnData{
+		Name: transaction.STORAGESC_STAKE_POOL_LOCK,
+		InputArgs: &stakePoolRequest{
+			ProviderID:   providerId,
+			ProviderType: providerType,
+		},
+	}, lock, fee, false, client...)
+}
+
+func StorageSCStakePoolUnlock(providerId string, providerType Provider, fee uint64, client ...string) (hash, out string, nonce int64, txn *transaction.Transaction, err error) {
+	return transaction.SmartContractTxnValueFee(StorageSmartContractAddress, transaction.SmartContractTxnData{
+		Name: transaction.STORAGESC_STAKE_POOL_UNLOCK,
+		InputArgs: &stakePoolRequest{
+			ProviderID:   providerId,
+			ProviderType: providerType,
+		},
+	}, 0, fee, false, client...)
+}
+
+func StorageSCCollectReward(providerId string, providerType Provider, fee uint64, client ...string) (hash, out string, nonce int64, txn *transaction.Transaction, err error) {
```

Comment:
Make sure you fix CLIs for it.

Reply:
this method isn't used anywhere searched in zboxcli and also zboxcli builds with this gosdk.

Reply:
Don't recreate it independently. You can build on the exisiting function so that updates to them are made at the same time.